src/tokenFactory/tokensFactory.ts

```diff
@@ -210,7 +210,8 @@ function createPresentationRequest(kid, iss, context, procUrl, procHash, data, e
 }
 
 function PSMHash(web3, jwt, did){
-	let json = jwt.concat(did);
+  let proxyAddress = did.split(':')[4];
```

Comment:
Hola!

Los cambios son correctos, pero la línea
```js
X.split(':')[4]
```
se repite todo el rato, así que creo que sería más cómodo tener una función pública auxiliar que haga solo eso, y que en el código, en lugar de hacer el split, llame a esa función.

La función puede ser algo como `getProxyAddresFromDID`

Adicionalmente, no sé cómo estarás probando estos cambios, pero lo que solemos hacer es trabajar en paralelo con los "example", que por lo que he visto, ya estáis en ello (https://github.com/alastria/alastria-identity-example/issues/45).

¡Buen trabajo! Y cualquier duda, aquí estamos :smiley:

Reply:
Se implementará en otro issue!